onnxruntime/contrib_ops/rocm/bert/fast_gelu_impl.cu

```diff
@@ -160,13 +275,26 @@ template <>
 bool LaunchFastGeluKernel(const hipDeviceProp_t& prop, hipStream_t stream, int input_length, int bias_length,
                           const half* input, const half* bias, half* output, bool use_half2) {
   constexpr int blockSize = 256;
-  if (use_half2 && prop.major >= 7 && (0 == (bias_length % 4) || 0 == (bias_length & 1))) {
+  if (use_half2 && prop.major >= 7 && (0 == (bias_length % 8) || 0 == (bias_length % 4) || 0 == (bias_length & 1))) {
     const half2 A2 = __float2half2_rn(A);
     const half2 B2 = __float2half2_rn(B);
     const half2 C2 = __float2half2_rn(C);
     const half2 one2 = __float2half2_rn(one);
     const half2 two2 = __float2half2_rn(two);
-    if (0 == (bias_length % 4)) {
+    if (0 == (bias_length % 8) && (input_length >= 3145728)) { // 3145728=8*128*3072
+      const int n = input_length / 8;
+      const int gridSize = (n + blockSize - 1) / blockSize;
+      const float4* input8 = reinterpret_cast<const float4*>(input);
+      const float4* bias8 = reinterpret_cast<const float4*>(bias);
+      float4* output8 = reinterpret_cast<float4*>(output);
+      unsigned int shmem = sizeof(half2) * 5;
+      if (bias == nullptr)
+        hipLaunchKernelGGL(HIP_KERNEL_NAME(FastGeluKernel8<blockSize>), dim3(gridSize), dim3(blockSize), 0,
+                           stream, A2, B2, C2, one2, two2, n, input8, output8);
+      else
+        hipLaunchKernelGGL(HIP_KERNEL_NAME(FastGeluKernel8Bias<blockSize>), dim3(gridSize), dim3(blockSize), 0,
+                           stream, A2, B2, C2, one2, two2, n, bias_length / 8, input8, bias8, output8);
+    } else if (0 == (bias_length % 4)) {
```

Comment:
If the below logic correct? I assume there are three cases (0 == (bias_length % 4), (0 == (bias_length % 2), and else, is this correct? I don't see to see (0 == (bias_length % 2) below.

Do we currently have test cases respectively for the three cases?

Reply:
The case (0 == (bias_length & 1)) has been included in [line 278](https://github.com/microsoft/onnxruntime/pull/11491/files#diff-c548bad1c014f26c9018ed219c43446dc33b93a152c68e90059db31507d88b71R278).

The test cases for hidden_size with size of multiples of 8, 4, and 2 will need to be added accordingly. [The current unit tests](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/contrib_ops/fastgelu_op_test.cc#L193) only have (batch_size, sequence_length, hidden_size) = (1, 2, 4).